### PR TITLE
fix(simple-rest): fix generated url with query string

### LINKED ISSUE #5359: [DOC] I want to remove ?& when calling API in useTable
### Documentation issue

I want to remove ?& when calling API in useTable

### Describe the thing to improve

I want to remove the ?& when calling the api in useTable and use the custom query I created.

### Describe the solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by amanzrx4**
fix: `dataProvider.getList` method adds `?` to the request url when there is no `filters`, `sorters` and `pagination` in the request. #5359

```diff
- Expected
+ Received

- "https://api.fake-rest.refine.dev/categories",
+ "https://api.fake-rest.refine.dev/categories?",
```

From now on, `dataProvider.getList` method will not add `?` to the request url when there is no `filters`, `sorters` and `pagination` in the request.

### Test plan (required)

image

### Closing issues 

`closes #5359` 

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

**Review by BatuhanW — CHANGES_REQUESTED** [review]
Hello @amanzrx4 thanks for the PR!

Could you add changeset? https://refine.dev/docs/further-readings/contributing/#changeset

**Comment by amanzrx4** [comment]
I've added. Could you check @BatuhanW

File: .changeset/perfect-ties-provide.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/simple-rest": major
```
Comment (BatuhanW) [inline .changeset/perfect-ties-provide.md:2]:
```suggestion
"@refinedev/simple-rest": patch
```

**Review by BatuhanW — CHANGES_REQUESTED** [review]

**Comment by amanzrx4** [comment]
I've also updated the fn implementation [here](https://github.com/refinedev/refine/blob/77c5cc9962a92a6d82d5ae7b7db716c4eae13286/packages/utils/index.ts)

**Comment by BatuhanW** [comment]
> @BatuhanW I've also updated the fn implementation [here](https://github.com/refinedev/refine/blob/77c5cc9962a92a6d82d5ae7b7db716c4eae13286/packages/utils/index.ts)

Thanks @amanzrx4! Looks great! We'll review it soon! Assigned it to January release.

There are 2 minor updates:
- Add a test for the new utility file
- Add new file to `refine.config.js`. https://github.com/refinedev/refine/blob/7c8827b43d9e378818be6ee23032925c97ce02d5/packages/simple-rest/refine.config.js#L18

We need to add this file to `refine.config.js`, because if someone swizzles this data provider, it won't copy this file to users' directory.

File: packages/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { stringify, ParsedUrlQueryInput } from "querystring";
```
Comment (aliemir) [inline packages/utils/index.ts:1]: We use the `query-string` package instead of `querystring`

File: .changeset/perfect-ties-provide.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/simple-rest": patch
+---
+
+add a urlWithQueryString util
```
Comment (aliemir) [inline .changeset/perfect-ties-provide.md:5]: Can we add more info about the issue the changes are going to resolve. This is going to be included in the changelogs and release notes for the next version.

**Review by aliemir — CHANGES_REQUESTED** [review]
Left two comments, should be easy to update. Thank you for the contribution.

**Review by alicanerdurmaz — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]
Great 🙌🏼

**Review by aliemir — APPROVED** [review]

**Pull request merged**